### PR TITLE
Bugfix updates for MacOS

### PR DESCRIPTION
**Pull request by ejfitzgerald**
Latest `develop` branch doesn't build and complete all tests in MacOS. This is due to the following:

1. A compilation warning, being treated as an error
2. Once compiling, the T-SNE test does not execute successfully under MacOS.

On the compilation side, the error is just a classic Mac weird `std::size_t` issue. On the test side I have not really investigated but the actual numerical result differs wildly from the expectation.

I would suggest that this probably warrants further investigation. In that light for the meantime, I have disabled this test.

Hopefully, by the end of next week we should have the CI-Mac attached to the Jenkins server so it will highlight these issues at PR time.

### CODE REVIEW

**Comment by troels-im** [comment]
We need to get that Mac mini up running.

**Comment by ejfitzgerald** [comment]
> We need to get that Mac mini up running.

Agreed

**Comment by chr15murray** [comment]
On my list for this coming week on the new Jenkins install

**Comment by ejfitzgerald** [comment]
Wow - we all need to get lives if we are answering GitHub comments on the weekend 😀

File: libs/ml/tests/ml/clustering/tsne_tests.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ using MyTypes = ::testing::Types<fetch::math::Tensor<float>, fetch::math::Tensor
 
 TYPED_TEST_CASE(TsneTests, MyTypes);
 
-TYPED_TEST(TsneTests, tsne_test_2d)
+TYPED_TEST(TsneTests, DISABLED_tsne_test_2d)
```
Comment (troels-im) [inline libs/ml/tests/ml/clustering/tsne_tests.cpp:37]: Probably should open a ticket for this if you haven't already.

**Review by troels-im — APPROVED** [review]
Open a ticket for the disabled test - rest looks good.

File: libs/ml/tests/ml/clustering/tsne_tests.cpp (hunk 1 above)
Comment (ejfitzgerald) [reply libs/ml/tests/ml/clustering/tsne_tests.cpp:37]: Done!

**Pull request merged**